File: exercises/ex1_play/play.f90
program play
    implicit none

    integer :: i
    integer, dimension(10) :: data
    integer :: int_var
    real :: real_var
    double precision :: double_var
    logical :: logical_var

    ! Populate the array with values from 1 to 10
    do i = 1, 10
        data(i) = i
    end do

    ! Assign values to variables
    int_var = 5
    real_var = 3.14
    double_var = 2.718281828459045d0
    logical_var = .true.

    call multiply_by_two(int_var)

    ! Check the logical variable and print if true
    if (logical_var) then
        print *, 'The logical variable is true.'
    end if

    ! Print all variables
    print *, 'Data array: ', data
    print *, 'Integer variable: ', int_var
    print *, 'Real variable: ', real_var
    print *, 'Double precision variable: ', double_var
    print *, 'logical variable: ', logical_var

    contains

    subroutine multiply_by_two(var)
        integer, intent(inout) :: var

        var = 2 * var
    end subroutine multiply_by_two

end program play

File: exercises/ex1_play/play.cpp
#include <iostream>
#include <array>

void multiply_by_two(int &var) {
    var = 2 * var;
}

int main() {
    std::array<int, 10> data;
    int int_var;
    float real_var;
    double double_var;
    bool logical_var;

    // Populate the array with values from 1 to 10
    for (int i = 0; i < 10; ++i) {
        data[i] = i + 1;
    }

    // Assign values to variables
    int_var = 5;
    real_var = 3.14f;
    double_var = 2.718281828459045;
    logical_var = true;

    multiply_by_two(int_var);

    // Check the logical variable and print if true
    if (logical_var) {
        std::cout << "The logical variable is true." << std::endl;
    }

    // Print all variables
    std::cout << "Data array: ";
    for (const auto &val : data) {
        std::cout << val << " ";
    }
    std::cout << std::endl;
    std::cout << "Integer variable: " << int_var << std::endl;
    std::cout << "Real variable: " << real_var << std::endl;
    std::cout << "Double precision variable: " << double_var << std::endl;
    std::cout << "Logical variable: " << std::boolalpha << logical_var << std::endl;

    return 0;
}
